rhtap/common.sh

```diff
@@ -20,6 +20,35 @@ timestamp() {
     date -u +"%Y-%m-%dT%H:%M:%SZ"
 }
 
+prepare-registry-user-pass() {
+    local image_registry="$1"
+    #
+    # Check if the IMAGE_REGISTRY_USER and IMAGE_REGISTRY_PASSWORD are set and if not
+    # compute the values from the image name (backward compitable with prior naming).
+    #
+    # Users should set IMAGE_REGISTRY_USER and IMAGE_REGISTRY_PASSWORD for the registry.
+    # For backwards compatibility use the ARTIFACTORY or NEXUS or QUAY creds in place
+    # and this code will determine which one to use.
+    #
+    if [[ -z "${IMAGE_REGISTRY_USER-""}" || -z "${IMAGE_REGISTRY_PASSWORD-""}" ]]; then
```

Comment:
It's a good question. This is to avoid throwing an error if this script is run with `set -u`, (which is recommended).

With `set -u`  enabled, any reference of an undefined var will immediately cause bash to throw an error. The ${VARNAME-"default"} syntax specifies a default value to use if the var doesn't exist, so it we ensure bash will do what we want when the var is not present.